src/Database/Query.php

```diff
@@ -881,6 +881,50 @@ public function where($conditions = null, $types = [], $overwrite = false)
         return $this;
     }
 
+    /**
+     * Adds a NOT NULL or IS NULL expression to the query
+     *
+     * @param array $fields A list of fields that should be not null
+     * @param bool $isNull Toggles between NOT NULL and NULL checks
+     * @return $this
+     */
+    protected function whereNullOrNotNull($fields, $isNull = true)
```

Comment:
This method is confusing, I'd rather not have it

Reply:
I think I would prefer having more code, yes.

Reply:
Would there be much duplicate code though? Couldn't `whereNull()` become:

```php
function whereNull($fields)
{
  $exp = $this->newExpr();
  // Could skip this if we want to require an array.
  if (is_array($fields) {
    $fields = [$fields];
  }
  foreach ($fields as $field) {
     $exp->isNull($field);
  }
  return $this->where($exp);
}
```

Reply:
Actually I would prefer everywhere in the framework methods that don't accept lots of different types.